src/modules/navigator/mission.cpp

```diff
@@ -953,10 +951,10 @@ Mission::altitude_sp_foh_update()
 		* The setpoint is set linearly and such that the system reaches the current altitude at the acceptance
 		* radius around the current waypoint
 		**/
-		float delta_alt = (get_absolute_altitude_for_item(_mission_item) - _mission_item_previous_alt);
+		float delta_alt = (get_absolute_altitude_for_item(_mission_item) - pos_sp_triplet->previous.alt);
```

Comment:
Actually what's happening is everytime navigator switches to manual/stab/altctl/etc it sets prev/curr/next to invalid. So the next time mission activates there's no FOH.
Although reading the code, it's not clear that's actually intended. I'll make it a bit more obvious.
https://github.com/PX4/Firmware/blob/master/src/modules/navigator/navigator_main.cpp#L616

Do you still think we should create a valid previous based on the position you entered mission?